core/trino-spi/src/main/java/io/trino/spi/block/Block.java

```diff
@@ -73,6 +74,14 @@ default Slice getSlice(int position, int offset, int length)
         throw new UnsupportedOperationException(getClass().getName());
     }
 
+    /**
+     * Gets a slice at {@code offset} in the value at {@code position} into the {@code output} slice output.
+     */
+    default void getSlice(int position, int offset, int length, SliceOutput output)
```

Comment:
This might be better named `writeSliceTo`, as that matches `writeBytesTo` and makes the calling code easier to read. The `get` name makes it appear that the value is discarded. Here's a changed caller in the next commit with the current naming:
```diff
- block.writeBytesTo(position, 0, entryLength, out);
+ block.getSlice(position, 0, entryLength, out);
```
With the proposed name:
```diff
- block.writeBytesTo(position, 0, entryLength, out);
+ block.writeSliceTo(position, 0, entryLength, out);
```